arrow-fx/src/test/kotlin/arrow/fx/ResourceTest.kt

```diff
@@ -59,16 +55,18 @@ class ResourceTest : UnitSpec() {
 private fun Resource.Companion.eqK() = object : EqK<ResourcePartialOf<ForIO, Throwable>> {
   override fun <A> Kind<ResourcePartialOf<ForIO, Throwable>, A>.eqK(other: Kind<ResourcePartialOf<ForIO, Throwable>, A>, EQ: Eq<A>): Boolean =
     (this.fix() to other.fix()).let {
-      val ls = it.first.invoke { IO.just(1) }.fix()
-      val rs = it.second.invoke { IO.just(1) }.fix()
+      val ls = it.first.use { IO.just(it) }.fix().attempt()
+      val rs = it.second.use { IO.just(it) }.fix().attempt()
       val compare = IO.applicative().mapN(ls, rs) { (l, r) -> l == r }.fix()
 
-      compare.unsafeRunTimed(5.seconds) == Some(true)
+      compare.unsafeRunTimed(10.seconds) == Some(true)
     }
 }
 
 private fun Resource.Companion.genK() = object : GenK<ResourcePartialOf<ForIO, Throwable>> {
-  override fun <A> genK(gen: Gen<A>): Gen<Kind<ResourcePartialOf<ForIO, Throwable>, A>> = gen.map {
-    Resource.just(it, IO.bracket())
-  }
+  override fun <A> genK(gen: Gen<A>): Gen<Kind<ResourcePartialOf<ForIO, Throwable>, A>> =
+    Gen.oneOf(
+      gen.map { just(it, IO.bracket()) },
+      IO.genK().genK(gen).map { it.liftF(IO.bracket()) }
```

Comment:
We should invoke all cases of the `sealed class Resource` here, no?

Reply:
There are 2 and the third one is `flatMap` which is already tested through, so I don't think we need it here, no?

Reply:
Here you can see how the `Gen` is build for `Resource` in Cats effect.

https://github.com/typelevel/cats-effect/blob/master/laws/shared/src/main/scala/cats/effect/laws/discipline/arbitrary.scala#L118

It build `Allocate` from `Gen<Kind<F, A>>` and `Gen<Kind<F, Unit>>`, it builds `FlatMap` from the allocate `Gen` and another for `Suspend`.

The idea here is that if your `Kind<F, A>` for example `IO<A>` is a strong generator that also generates `IO` at depths etc so it can cover all its cases.